packages/client/src/stripHeaders.ts

```diff
@@ -0,0 +1,41 @@
+import { HTTPHeaders } from './links/core';
+
+// from https://github.com/nodejs/undici/blob/9e5316c8b04a7b35522d0d5b8de71f67fa2a3c34/lib/fetch/constants.js
+const forbiddenHeaderNames = [
+  'accept-charset',
+  'accept-encoding',
+  'access-control-request-headers',
+  'access-control-request-method',
+  'connection',
+  'content-length',
+  'cookie',
+  'cookie2',
+  'date',
+  'dnt',
+  'expect',
+  'host',
+  'keep-alive',
+  'origin',
+  'referer',
+  'te',
+  'trailer',
+  'transfer-encoding',
+  'upgrade',
+  'via',
+];
+
+/**
+ * Strips an HTTP header object of internally reserved keys (for undici/Node.js built-in fetch since v18)
+ * @param headers Your HTTP header function/object
+ * @returns Stripped headers without blacklisted keys
+ */
+export const stripHeaders = async (
+  headers: HTTPHeaders | (() => HTTPHeaders | Promise<HTTPHeaders>),
```

Comment:
Simplify

````suggestion
  headers: HTTPHeaders,
```

Reply:
@KATT Related to the convo above. Should we keep this function as flexible as possible regardless of individual developer preference?

Reply:
No. Keep it minimal and stupid. If people want something else they can roll their own wrapper